worlds/ffmq/__init__.py

```diff
@@ -204,16 +205,14 @@ def extend_hint_information(self, hint_data):
                             for location in exit_check.connected_region.locations:
                                 if location.address:
                                     hint = []
-                                    if self.multiworld.map_shuffle[self.player] != "dungeons":
+                                    if self.options.map_shuffle != "dungeons":
                                         hint.append((subregion.split("Subregion ")[-1] + (" Region" if subregion not
                                                     in single_location_regions else "")))
-                                    if self.multiworld.map_shuffle[self.player] != "overworld" and subregion not in \
-                                            ("Subregion Mac's Ship", "Subregion Doom Castle"):
+                                    if self.options.map_shuffle != "overworld":
```

Comment:
Just checking that this subregion removal is intended here.

Reply:
> Just checking that the removal of the subregion check here is intended.

From the description of this PR:

> Entrance hints will now show locations at Mac's Ship (why did I go out of my way to exclude it? It is a mystery)

Reply:
And I wasn't sure if Doom Castle was relevant to that. But by your answer, I'll just assume it was